pilot/pkg/xds/ads.go

```diff
@@ -168,6 +169,11 @@ func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.Di
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
+
+	if s.InternalGen != nil && req.TypeUrl == health.HealthInfoTypeURL {
```

Comment:
Register the HealthInfoTypeURL  generator as others, and handle it similar to others